aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,172 @@
+
+![AOT mode](index/_static/aotmode.png)
+
+This is similar to how the [Ngen.exe tool](/dotnet/framework/tools/ngen-exe-native-image-generator) has historically allowed AOT compilation of .NET binaries into native machine code. More recently, CoreRT provides a complete native AOT .NET runtime for executing .NET binaries.
+
+## Interpreted versus AOT
+
+Interpreted mode provides a faster development cycle than AOT. When code is changed under the interpreted model, the app can be rebuilt and reloaded in the browser within a few seconds. An AOT rebuild might take several minutes to compile and reload. Because interpreted mode offers faster recompilation, it may be preferred for development. It remains an open question how to best apply interpreted and AOT modes of compilation in Blazor.
+
+## Browsers that don't support WebAssembly
+
+When the .NET runtime is compiled into WebAssembly, it's also compiled into [asm.js](https://en.wikipedia.org/wiki/Asm.js). *asm.js* is a subset of JavaScript and can be executed by JavaScript runtimes in browsers going back several years. When Blazor loads in the browser, it checks for WebAssembly support. If WebAssembly isn't supported, the *asm.js* runtime is loaded. *asm.js* isn't always used becuase it's larger and slower than the WebAssembly runtime.
+
+## Blazor, a SPA framework
+
+Blazor has a growing standard feature set to solve common app requirements, such as UI composition, state management, and routing. Features are designed around the strengths of .NET and the C# language with careful consideration given to tooling support.
+
+Blazor is inspired by the most popular SPA frameworks, such as [React](https://reactjs.org/), [Vue](https://vuejs.org/), and [Angular](https://angular.io/), plus some Microsoft technologies, such as [Razor Pages](xref:mvc/razor-pages/index). The goal is to provide SPA features that web developers have found most successful in existing frameworks in a way that capitalizes on .NET's strengths.
+
+## Blazor components
+
+In SPA frameworks, apps are built with *components*. A component usually represents a piece of UI, such as a page, dialog, or data entry form. Components can be nested, reused, and shared between projects.
+
+In Blazor, a component is a .NET class. The class can either be written directly, as a C# class (*\*.cs*), or more commonly in the form of a Razor markup page (*\*.cshtml*).
+
+Many design patterns are possible using [Razor](xref:mvc/views/razor) as a foundation for Blazor. Razor, which has been around since 2010, is a syntax for combining HTML markup with C# code. Razor is designed for developer productivity, allowing the developer to switch between markup and C# in the same file with IntelliSense support. The following markup is an example of a basic custom dialog component in a Razor file:
```

Comment:
Let's not call out Razor's history and instead focus on defining what Razor is:

[Razor](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/razor) is a syntax for combining HTML markup with C# code. Razor . . .